lib/stdlib/src/sys.erl

```diff
@@ -510,27 +510,41 @@ do_get_state(Mod, Misc) ->
     end.
 
 do_replace_state(StateFun, Mod, Misc) ->
-    case erlang:function_exported(Mod, system_replace_state, 2) of
-	true ->
-	    try
-		{ok, State, NMisc} = Mod:system_replace_state(StateFun, Misc),
-		{State, NMisc}
-	    catch
-		Cl:Exc ->
-		    {{error, {callback_failed,{Mod,system_replace_state},{Cl,Exc}}}, Misc}
-	    end;
-	false ->
-	    try
-		NMisc = StateFun(Misc),
-		{NMisc, NMisc}
-	    catch
-		Cl:Exc ->
-		    {{error, {callback_failed,StateFun,{Cl,Exc}}}, Misc}
-	    end
-    end.
+Sensitivity = erlang:process_flag(sensitive, false),
```

Comment:
Disabling the sensitive flag here exposes the process information until you set the sensitive flag back to what it was. Meaning one can look into the sensitive process data by concurrently calling `replace_state` and `process_info`.

Reply:
Yeah. I think we need to fetch this flag without changing it.